modules/apps/commerce/commerce-frontend-impl/src/main/java/com/liferay/commerce/frontend/internal/search/CommerceSearchResource.java

```diff
@@ -175,20 +173,16 @@ private String _getAccountManagementPortletEditURL(
 	}
 
 	private SearchItemModel _getSearchItemModel(
-			CPCatalogEntry cpCatalogEntry, ThemeDisplay themeDisplay)
+			CPCatalogEntry cpCatalogEntry, ThemeDisplay themeDisplay,
+			long commerceAccountId)
```

Comment:
Hi @CrescenzoRegaLF I remember you already explained me, but I cannot remember why you did this change, I think it should work also with the old version, can you explain it again?

Reply:
The problem was that the method was retrieving the commerceAccountId from the context, like this
`CommerceUtil.getCommerceAccountId ((CommerceContext) httpServletRequest.getAttribute (CommerceWebKeys.COMMERCE_CONTEXT))`
which however returned the value 0

Reply:
Ok but that is only the account is signed in, it also return -1 in guest account case, are you sure it never return -1?